File: scripts/triggers/synchronator.py
############## BEGIN CONFIGURATION ##############

command_code="power"
command_parameter="deviceon"

############### END CONFIGURATION ###############

import sysv_ipc

msq_key=sysv_ipc.ftok('/tmp/synchronator.msq', 66, silence_warning = True)
msq  = sysv_ipc.MessageQueue(msq_key)
msq_command=command_code + "=" + command_parameter + "\0"
msq.send(msq_command)

File: scripts/triggers/synchronator.pl
#!/usr/bin/perl
############## BEGIN CONFIGURATION ##############

$command_code="power";
$command_parameter="deviceon";

############### END CONFIGURATION ###############

use IPC::SysV qw(S_IRUSR);
use IPC::SysV qw(ftok);

$msq_key = ftok("/tmp/synchronator.msq", 'B');
$msq = msgget($msq_key, S_IRUSR);
defined($msq) || die "msgget failed: $!";
$msq_command = $command_code . '=' . $command_parameter . "\0";
$type_sent = 1;
msgsnd($msq, pack("l! a*", $type_sent, $msq_command), 0) || die "msgsnd failed: $!";

exit;